test/assistant-app-test.js

```diff
@@ -159,4 +161,101 @@ describe('AssistantApp', function () {
         .equal(true);
     });
   });
+
+  describe('#handleRequest', function () {
+    let mockRequest;
+    let app;
+
+    beforeEach(function () {
+      mockRequest = new MockRequest(headerV2, JSON.parse(JSON.stringify(apiAiAppRequestBodyNewSessionMock)));
+      app = new ApiAiApp({request: mockRequest, response: mockResponse});
+    });
+
+    it('Should resolve a promise when actionMap contains a handler that returns a promise', function (done) {
+      let handler = app => {
+        return Promise.resolve('success');
+      };
+
+      const actionMap = new Map();
+      actionMap.set(mockRequest.body.result.action, handler);
```

Comment:
This is mostly a nit, but you might be able to get around importing `ApiAiApp` (or `DialogflowApp` now) and using `mockRequest.body.result.action` by mocking a `AssistantApp` `getIntent()` method here, since line 544 in `handleRequest()` just uses this to find the right handler function.

Reply:
This was done, thanks :)